public/js/index.js

```diff
@@ -1567,10 +1638,14 @@ function importFromUrl (url) {
 
 // mode
 ui.toolbar.mode.click(function () {
```

Comment:
introduce a function `isAllowUserChangeMode`
```js
function isAllowUserChangeMode () {
  const isOwner = personalInfo.userid && window.owner && personalInfo.userid === window.owner
  return isOwner || !blockSourceView
}
```

and mode select logic would be

```js
if (isAllowUserChangeMode()) {
    return toggleMode()
}
```